src/js/relay.firefox.com/get_profile_data.js

```diff
@@ -1,419 +1,453 @@
-        if (isValid) {
+          // Save new alias label
+          const updatedRelayAddress = localStorageRelayAddresses.filter(
+            (address) => address.id == aliasId
+          )[0];
+          updatedRelayAddress.description = newAliasLabel;
+          browser.storage.local.set({
+            relayAddresses: localStorageRelayAddresses,
+          });
+
+          // show placeholder text if the label is blank
+          if (aliasLabelInput.value === "") {
+            aliasLabelWrapper.classList.remove("user-created-label");
+            aliasLabelInput.placeholder = defaultAliasLabelText;
+          } else {
+            aliasLabelWrapper.classList.add("user-created-label");
+            aliasLabelWrapper.classList.add("show-saved-confirmation");
+          }
+
+          aliasLabelInput.dataset.label = newAliasLabel;
+          setTimeout(() => {
+            aliasLabelWrapper.classList.remove("show-saved-confirmation");
+          }, 1000);
+        };
+        aliasLabelInput.addEventListener("focusout", () => {
+          const isValid = aliasLabelForm.reportValidity();
+          if (isValid) {
+            saveAliasLabel();
+          }
+        });
+        aliasLabelForm?.addEventListener("submit", (event) => {
+          event.preventDefault();
           saveAliasLabel();
-        }
-      });
-      aliasLabelForm?.addEventListener("submit", (event) => {
-        event.preventDefault();
-        saveAliasLabel();
-        aliasLabelInput.blur();
-      });
+          aliasLabelInput.blur();
+        });
 
-      // Get and store the relay addresses from the account profile page,
-      // so they can be used later, even if the API endpoint is down
+        // Get and store the relay addresses from the account profile page,
+        // so they can be used later, even if the API endpoint is down
 
-      const relayAddress = {
-        id: aliasId,
-        address: aliasCardData.relayAddress,
-        description: storedAliasLabel,
-        generated_for: storedAliasGeneratedFor,
-      };
+        const relayAddress = {
+          id: aliasId,
+          address: aliasCardData.relayAddress,
+          description: storedAliasLabel,
+          generated_for: storedAliasGeneratedFor,
+        };
 
-      localStorageRelayAddresses.push(relayAddress);
+        localStorageRelayAddresses.push(relayAddress);
+      }
+
+      if (localStorageRelayAddresses.length === 0) {
+        // If we weren't able to scrape alias data from the page, that means
+        // the React version of the website (with a different DOM structure)
+        // has been deployed. Since the React version handles local storage of
+        // label data by itself, we don't need to add our own listeners,
+        // and we can just fetch label data from the API.
+        const serverRelayAddresses = await apiRequest(apiRelayAddressesURL);
+        localStorageRelayAddresses.push(...serverRelayAddresses);
+      }
```

Comment:
This part is also new, and is part of the second change (54e5c27666a6021ff86d08fc3954179b4fc23802): whereas we scrape the DOM to get aliases and their labels with the old website, we just fetch them from the API in the React-based version.